sea-orm-macros/src/derives/simple_model/column.rs

```diff
@@ -0,0 +1,85 @@
+use heck::{CamelCase, MixedCase, SnakeCase};
+use proc_macro2::{Ident, TokenStream};
+use quote::{format_ident, quote};
+use syn::{punctuated::Punctuated, token::Comma, Field, Result, Visibility};
+
+pub(crate) fn expand_column(
+    vis: Visibility,
+    ident: Ident,
+    fields: Punctuated<Field, Comma>,
+) -> Result<TokenStream> {
+    let column_ident = format_ident!("{}Column", ident);
+    let entity_ident = format_ident!("{}Entity", ident);
+
+    let column_fields = fields.iter().map(|field| {
+        format_ident!(
+            "{}",
+            field.ident.clone().unwrap().to_string().to_camel_case()
+        )
+    });
+    let column_fields_cloned = column_fields.clone();
+    let column_field_names = fields
+        .iter()
+        .map(|field| field.ident.clone().unwrap().to_string());
+
+    let column_from_str_fields = fields.iter().map(|field| {
+        let field_camel = format_ident!(
+            "{}",
+            field.ident.clone().unwrap().to_string().to_camel_case()
+        );
+        let column_str_snake = field_camel.to_string().to_snake_case();
+        let column_str_mixed = field_camel.to_string().to_mixed_case();
+        quote!(
+            #column_str_snake | #column_str_mixed => Ok(#column_ident::#field_camel)
+        )
+    });
+
+    let expanded = quote!(
+        #[derive(Copy, Clone, Debug, sea_orm::sea_strum::EnumIter)]
+        #vis enum #column_ident {
+            #(#column_fields),*
+        }
+
+        impl #column_ident {
+            fn default_as_str(&self) -> &str {
+                match self {
+                    #(Self::#column_fields_cloned => #column_field_names),*
+                }
+            }
+        }
+
+        impl sea_orm::entity::ColumnTrait for #column_ident {
+            type EntityName = #entity_ident;
+
+            fn def(&self) -> sea_orm::entity::ColumnDef {
+                // TODO: Generate column def
+                panic!("No ColumnDef")
```

Comment:
Generating the column def might be a little difficult. Might have to create mapping from types to `ColumnType`.